File: 2024/day1.2.dart
import 'util.dart';

void main() {
  var fileContent = readFile('./files/day1.txt');
  var lines = fileContent.split('\n');
  var total = 0;

  for (final line in lines) {
    var count = 0;
    var items = line.split('   ');
    var left = int.parse(items[0]);

    for (var i = 0; i < lines.length; i++) {
      if (left == int.parse(lines[i].split('   ')[1])) count++;
    }
    total += left * count;
  }

  print(total);
}

File: 2022/2022/day1/day1.2.swift
//
//  day1.2.swift
//  2022
//
//  Created by Teboho Mphure on 2022/12/01.
//

import Foundation

class Day1_2 {
    var total = 0
    var totalCalories: [Int] = []
    
    func GetTopThree() {
        let lines = FileUtils.ReadFile(directory: "repos/advent-of-code/2022/2022/files/day1.txt")

        for line in lines {
            let num = Int(line) ?? 0
            
            if num != 0 {
                total += num
            } else {
                totalCalories.append(total)
                total = 0
            }
        }

        for _ in 1...3 {
            if let answer = totalCalories.max() {
                total += answer
                let index = totalCalories.firstIndex(of: answer)
                totalCalories.remove(at: index!)
            }
        }
        print(total)
        
    }
}
